openproblems/data/tabula_muris_senis.py

```diff
@@ -12,14 +13,34 @@
 METHOD_ALIASES = {"10x 3' v2": "droplet", "Smart-seq2": "facs"}
 
 
+def check_unknown_organs(datasets, organ_list):
+    known_organs = set([t["label"] for d in datasets for t in d["tissue"]])
+    unknown_organs = set(organ_list) - known_organs
+    if unknown_organs:
+        raise ValueError(
+            f"Unknown organs provided in `organ_list': {', '.join(unknown_organs)}. "
+            f"Known organs are {', '.join(known_organs)}"
+        )
+
+
 def matching_dataset(dataset, method_list, organ_list):
-    # we want to skip combined datasets
-    if len(dataset["tissue"]) > 1:
+    assert len(dataset["assay"]) == 1
+    method = dataset["assay"][0]["label"]
+    method = METHOD_ALIASES[method]
```

Comment:
Don't understand this entirely... at the moment all Tabula Muris Senis collection datasets are a single assay, but if in future they add datasets from 2 assays we wouldn't want to download them together even if we would potentially ask for both? Or is this because the `matching_dataset` method is specifically set up for single-dataset download only?

Reply:
Currently, all Tabula Muris Senis collection datasets have a single assay, either Smart-seq2 or 10x 3' v2. The current code makes a strong assumption that this is the case, and will break if there will appear a new 2 assay dataset. This is probably undesirable, I'll fix it.

A separate question is if we want to download multiple assay datasets if the use passes empty `method_list` and such datasets are available? Only if we assume that such a dataset will have all organs, or the requested organs. In theory, I can rewrite the code to first determine available assays/organ combinations and then resolve from there to the minimal number of datasets to download. However, I think it will overcomplicate the code.

What do you think?

Reply:
Let's just stick with downloading single assay/tissue datasets. Keeps things simple.

Reply:
I'm mainly bringing this up as I was hoping to use this data loader as a general template for CELLxGENE explore data loaders. Other datasets may not be single assay only (e.g., HLCA). How many dataset collection-specific stuff is currently in the data loader? Probably too much to make this general purpose, no?

Maybe given timeline you're right though, @scottgigante-immunai ... keep it simple for now. At least we have 1 version of a data loader from CELLxGENE explore.

Reply:
Yeah, I thought about asking to turn this into a more general utility function but that seems like a nice-to-have, and right now this is blocking everything